tox.ini

```diff
@@ -1,11 +1,48 @@
 # Note: Run "python bootstrap.py" before running Tox, to generate metadata.
-#
-# To run Tox against all supported Python interpreters, you can set:
-#
-# export TOXENV='py27,py3{3,4,5,6},pypy,pypy3'
+[tox]
+envlist = py27,
+          py33,
+          py34,
+          py35,
+          py36,
+          py37,
+          pypy,
+          pypy3,
+          coverage
+skip_missing_interpreters=true
 
 [testenv]
 deps=-rtests/requirements.txt
-passenv=APPDATA USERPROFILE HOMEDRIVE HOMEPATH windir APPVEYOR
-commands=py.test {posargs}
+setenv=COVERAGE_FILE={toxworkdir}/.coverage.{envname}
+passenv=APPDATA USERPROFILE HOMEDRIVE HOMEPATH windir APPVEYOR APPVEYOR_* CI CODECOV_* TRAVIS TRAVIS_*
```

Comment:
I suspect CODECOV_* is for code coverage but the addition of these other variables is more boilerplate. Is there someplace this boilerplate is maintained? I've been thinking of merging /jaraco/skeleton into this project, but have mainly resisted as this project can't use setuptools_scm.

Reply:
Some fraction of these extra variables are necessary for `codecov` to detect that the run was on Travis or Appveyor (I think at *least* `CI` and `APPVEYOR` are needed) - if it doesn't detect that, it requires a user upload token and will fail to upload.

To be honest, I just started copying over the values from [dateutil's `tox.ini`](https://github.com/dateutil/dateutil/blob/master/tox.ini) until things worked. @gaborbernat slaved over the CI configuration getting all that stuff working for pretty much a full day when I ran a sprint in London and he actually knows `tox` in and out, so I was not amazingly confident straying from his example.

I think it might be a good idea to leave these as is for now, possibly with a comment that says, "These have been cargo-culted, we should actually go through and understand why each is necessary". As it is, the code coverage seems to look a bit different when it's *only* in a PR and not on master, so any experimentation might be better done on a PR against a master that already has codecov set up.

Reply:
I don't see `APPDATA`, `HOMEDRIVE`, `HOMEPATH` or `windir` being set in [dateutil's tox.ini](https://github.com/dateutil/dateutil/blob/master/tox.ini#L18).

Reply:
@benoit-pierre Those were already in the `setuptools` configuration, so I left them in place.

Reply:
Right! I see `windir` is referenced in `pkg_resources`. Not sure about the other ones, pip's tox.ini does not set any of those.